Bluesky.js

```diff
@@ -0,0 +1,147 @@
+{
+	"translatorID": "3bba003a-ad42-457e-9ea1-547df39d9d00",
+	"label": "Bluesky",
+	"creator": "Stephan Hügel",
+	"target": "^https://bsky\\.app/",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2024-12-06 20:35:13"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2024 Stephan Hügel <urschrei@gmail.com>
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes('/post/')) {
+		return 'forumPost';
+	}
+	return false;
+}
+
+async function doWeb(doc, url) {
+	await scrapeAPI(doc, url);
+}
+
+async function scrapeAPI(doc, url) {
+	const handle = /(?:\/profile\/)(([^/]+))/;
+	const postId = /(?:\/post\/)([a-zA-Z0-9]+)/;
+
+	let foundHandle = url.match(handle)[1];
+	let foundPostId = url.match(postId)[1];
+
+	let apiUrl = `https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread?uri=at://${foundHandle}/app.bsky.feed.post/${foundPostId}`;
+	let data = await ZU.requestJSON(apiUrl);
+	if (data.thread && data.thread.post) {
+		let post = data.thread.post;
+		let item = new Zotero.Item("forumPost");
+		// Main post details
+		// cut off titles longer than 140 characters
+		let title = post.record.text || "Bluesky Skeet";
+		if (title.length < 140) {
+			item.title = title;
+		}
+		else {
+			item.title = ZU.ellipsize(title, 140, true);
+			item.abstractNote = title;
+			// we'll keep the full text in the abstract in this case;
+		}
```

Comment:
@urschrei, I'm not a maintainer here, but my recommendation is to always populate item.abstractNote with the original title, even when item.title isn't abbreviated. That seems like the cleanest way for the ability to reliably cite the full text of a post. An alternative for citing is to cite using "item.abstractNote" and fall back to "item.title"  if the former isn't populated, but that assumes people didn't use "item.abstractNote" for another purpose for posts that didn't have this field populated.

https://github.com/zotero/translators/blob/f2ddb99a63b7ace233ecad1e0a33a0585b8ee7d2/Twitter.js#L108-L109 also removes newlines (and other white-space) from posts, which is not covered by the ellipsize function, and it probably makes sense to do that here as well. See also https://github.com/zotero/utilities/blob/c879fa18113646c18abc951ed86481bfc2b2d134/utilities.js#L991